### DISCUSSION TITLE
Use request variable in ref context

### DISCUSSION THREAD

**Discussion by wekempf** [AnWeber/httpyac#520] [General]
Is there a way to use a variable defined for one request in another request it references?

```
###
# @name a
GET /foo/{{id}}/bar

###
# @name b
# @ref a
@id={{$input input id? $value: 123}}
GET /baz/{{id}}
```

I need to provide the value when running `b` but the same value should be used when first running `a`.

**Comment by AnWeber** [comment]
Order is important:-)
In principle, the whole application actually works (with exceptions) that each line is interpreted into an instruction and it is executed in that order. Just change the order between the variable and the `@ref`. 

```
# @name a
GET https://postman-echo.com?foo={{id}}
###
@id={{$input input id? $value: 123}}
# @ref a
GET https://postman-echo.com?foo={{id}}
```

But I would generally advise against using it this way, and rather use a global variable instead. This has the advantage that both requests can run independently.

```
@id={{$input input id? $value: 123}}
###
# @name a
GET https://postman-echo.com?foo={{id}}
###
# @ref a
GET https://postman-echo.com?foo={{id}}
```
